extensions/amp-bind/0.1/bind-impl.js

```diff
@@ -100,9 +94,9 @@ export class Bind {
     };
 
     this.ampdoc.whenBodyAvailable().then(body => {
-      const {bindings, expressions} = this.scanForBindings_(body);
-      this.bindings_ = bindings;
-      this.evaluator_ = new BindEvaluator(expressions);
+      const {boundElements, evaluatees} = this.scanForBindings_(body);
```

Comment:
I think we need `ampdoc.whenReady()` here. Basically, `whenBodyAvailable` means when document's opening `<body>` has been parsed. `whenReady` means that final `</html>` has been parsed.

That being said, we need to ensure that we prioritize it properly - it's a lot of work to do CPU-wise and it'd be unideal if we fight over resources with initial rendering. Not sure how we can do this yet. I think it's ok to add a `TODO` and file an issue for now.